### ISSUE TITLE
Broken crosh

### ISSUE THREAD

**Issue by PoisonNinja** [CodeBuds/Cri#5]
The crosh that your script copies into /usr/bin is broken. It crashes as soon as I open it. The md5 sum of my stock crosh is b0d00afff680787e38fe86ce2d2d145a. The one that you copy to /usr/bin when uninstalling is 4122a6739f84be44f043b3c103d3f5b0.

**Comment by eliliam** [comment]
Where are you right now? I'll come show you the fix

**Comment by PoisonNinja** [comment]
I uninstalled it already. I'll test it out over the weekend again.

**Comment by smerkousdavid** [comment]
You can use a virtual terminal not crosh, by using the forward and backward
buttons with Ctrl+alt... In there try rerunning rootmount, and then the
second script. The only main reason it can crash is because when wget tries
to pull a file the doesn't exist it turns blank and crashes meaning that
your current internet connection is unstable, or an error moving our
modified crosh to the use/bin. Which could be permissions and you should
run rootmount.

**Comment by eliliam** [comment]
What's your email Jason?

**Comment by PoisonNinja** [comment]
jasonlu31415@gmail.com. @smerkousdavid root is already mounted rw. I have an internet connection. Also the script does 

``` Shell
sudo wget -q -O - https://raw.github.com/CodeBuds/Cri/master/Crosh > crosh
```

Why not do 

``` Shell
sudo wget -q -O crosh https://raw.github.com/CodeBuds/Cri/master/Crosh
```

The first code snippet says permission denied.

**Comment by eliliam** [comment]
Because we move what we download to replace the crosh file, it's more clear
the way we do it

**Comment by PoisonNinja** [comment]
``` Shell
bash: crosh: Permission denied
```

This is using the code from the uninstall script. Root verification is disabled, and the filesystem is mounted as RW. The reason why it says permission denied is because the redirect is done by the shell. You need to use this if you want to be able to see what your using the replace the crosh file:

``` Shell
sudo wget -q -O - https://raw.github.com/CodeBuds/Cri/master/Crosh | sudo tee crosh > /dev/null
```

See [here](http://stackoverflow.com/questions/82256/how-do-i-use-sudo-to-redirect-output-to-a-location-i-dont-have-permission-to-wr) for more information.

**Comment by smerkousdavid** [comment]
Both of those way are not the best way of writing to a file, because piping will grab the current buffer from the file even if it hasn't finished, so ctrl+c will corrupt the buffer and the file. wget has a builtin output file, so it buffers like a pipe but won't copy over until the installation is finished... You see when you put -O will place it into the buffer file that gets deleted afterwards, not a good idea.

Use this instead:
sudo wget -q --no-check-certificate "https://raw.github.com/CodeBuds/Cri/master/Crosh" -O /usr/bin/crosh

or even:

sudo su -c 'wget -q --no-check-certificate "https://raw.github.com/CodeBuds/Cri/master/Crosh" -O /usr/bin/criCrosh'

if [ -s /usr/bin/criCrosh ]; then
sudo su -c 'mv /usr/bin/criCrosh /usr/bin/crosh'
sudo su -c 'chmod 755 /usr/bin/crosh'
echo "Done..."
else
echo "File corrupted"
exit 1
fi

EDIT: Almost forgot chmod

**Comment by PoisonNinja** [comment]
I just ran your script. Crosh crashed as soon as I opened it.

**Comment by smerkousdavid** [comment]
Did it echo corrupted?

**Comment by PoisonNinja** [comment]
No. It said "Done...". I'll link a video
Edit: https://drive.google.com/file/d/0BzKnVWnfS942XzdzX0w1ZDNfVGc/view?usp=drivesdk

**Comment by smerkousdavid** [comment]
Can you email me the current code in /usr/bin/crosh

to move to downloads:
sudo su -c 'mv /usr/bin/crosh ~/Downloads'

**Comment by PoisonNinja** [comment]
I'll attach it.
[crosh.zip](https://github.com/CodeBuds/Cri/files/53275/crosh.zip)

**Comment by PoisonNinja** [comment]
I've been testing the different versions of crosh in your repo. The original crosh is already broken.

**Comment by eliliam** [comment]
No, the crosh in the repo is a fresh version.

**Comment by smerkousdavid** [comment]
Try this:

[crosh.txt](https://github.com/CodeBuds/Cri/files/53279/crosh.txt)
Couldn't upload zip so rename this

**Comment by PoisonNinja** [comment]
That works. However, I figured out what was causing your copy of crosh to crash. I'll upload the patched copy. It's attached as a zip file, and it's called croshtest. Make sure to rename it before copying it.
[croshtest.zip](https://github.com/CodeBuds/Cri/files/53280/croshtest.zip)

**Comment by PoisonNinja** [comment]
The one I linked above is the one available from your repo, so all the prompts are still your custom ones. Apparently, some lines were removed that shouldn't have been removed (I think it was the ones dealing with repl). Everything else is fine.

**Comment by smerkousdavid** [comment]
before we upload Eli needs to test this on his system... and make sure that dialog works

**Comment by eliliam** [comment]
Okay, give me a few minutes.

**Comment by oliverwarne** [comment]
Fixed by Jasons PR #6
